### ISSUE TITLE
Unable to open Zed.dmg

### ISSUE THREAD

**Issue by 1366613** [zed-industries/zed#7132]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I'm unable to open Zed.dmg file. Nothing happens after clicking on the file in Finder.

Running `file Zed.dmg` gives me `zlib compressed data`, where any other typical dmg file would give an output similar to `DOS/MBR boot sector; partition 1 : ID=0xee, start-CHS (0x3ff,254,63), end-CHS (0x3ff,254,63), startsector 1, 456703 sectors, extended partition table (last)`

Anyway, as I see, Homebrew was able to install Zed on my system using the dmg file it downloaded, so I'm not sure what's wrong in here.

### Environment

macOS Sonoma 14.4 Beta (23E5180j) on Apple Silicon M2

**Comment by mrnugget** [comment]
Which version? When I download `0.119.21` from https://zed.dev/download I get this screen when clicking in Finder:

![screenshot-2024-01-31-08 50 32@2x](https://github.com/zed-industries/zed/assets/1185253/c9f6bde4-7b2e-45ef-be3a-a4c519ddd1f4)

Does this match yours too?

```
φ file Zed.dmg
Zed.dmg: zlib compressed data
φ md5 Zed.dmg
MD5 (Zed.dmg) = 4fa304ead0698d49774ca0f260609111
φ ls -l Zed.dmg
-rw-r--r--@ 1 thorstenball  staff  115713508 Jan 31 08:49 Zed.dmg
```

**Comment by asrob** [comment]
`$ file Zed.dmg
Zed.dmg: zlib compressed data

$ md5 Zed.dmg
MD5 (Zed.dmg) = 4fa304ead0698d49774ca0f260609111

$ ls -l Zed.dmg
-rw-r--r--@ 1 asrob  staff  115713508 Jan 31 20:26 Zed.dmg`

I've installed Zed (Zed 0.119.21) a few minutes ago and it works well. I could open .dmg, like @mrnugget I got that screen too.

**Comment by 1366613** [comment]
@mrnugget Turns out this is a nasty bug in macOS Sonoma 14.4 Developer Beta. I've reported this to Apple. It turns out, macOS is unable to process any DMG files with the zlib compression format.

**Comment by mrnugget** [comment]
Very interesting, thanks for reporting back!

**Comment by GeVic** [comment]
> @mrnugget Turns out this is a bug in macOS Sonoma 14.4 Developer Beta. I've reported this to Apple. Apparently, the system is unable to process any DMG files with the zlib compression format.

Yeah, I can confirm this. Same issue with me.

**Comment by 1366613** [comment]
> > @mrnugget Turns out this is a bug in macOS Sonoma 14.4 Developer Beta. I've reported this to Apple. Apparently, the system is unable to process any DMG files with the zlib compression format.
> 
> same problem here. Unable to open the .dmg with Sonoma 14.4 :/ no way around the problem at this point in time it seems

One workaround for the time being is unpacking the dmg with (Pacifist)(https://www.charlessoft.com) and copying the app to the Applications folder.

**Comment by 1366613** [comment]
> > > > @mrnugget Turns out this is a bug in macOS Sonoma 14.4 Developer Beta. I've reported this to Apple. Apparently, the system is unable to process any DMG files with the zlib compression format.
> > > 
> > > 
> > > same problem here. Unable to open the .dmg with Sonoma 14.4 :/ no way around the problem at this point in time it seems
> > 
> > 
> > One workaround for the time being is unpacking the dmg file with [Pacifist](https://www.charlessoft.com) and copying the app to the Applications folder, or otherwise simply installing Zed with `brew install zed`
> 
> Thanks! worked like a charm, didnt even think to look if a brew is available :)

brew isn't installing the unstable versions.
Btw I've got official response from the Apple support team about the bug being fixed in build `23E5191e`.

**Comment by sarathjasrin** [comment]
It's working for me as well. My os version `14.5 (23F79)`. Installed via brew
